src/jit/codegenarmarch.cpp

```diff
@@ -3720,6 +3720,17 @@ void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pIni
         return;
     }
 
+#ifdef _TARGET_ARM64_
+    // For arm64, we are probing the frame *before* the callee-saved registers are saved. The 'initReg' might in fact
+    // have been calculated to be one of the callee-saved registers (say, if all the integer argument registers are
+    // in use, and perhaps with other conditions being satisfied). If that's true, then just use the REG_SCRATCH
+    // register instead. (We could probably just always use REG_SCRATCH.)
+    if ((genRegMask(initReg) & RBM_CALLEE_SAVED) != 0)
+    {
+        initReg = REG_SCRATCH;
+    }
+#endif // _TARGET_ARM64_
+
```

Comment:
I don't think that this is the best way to fix this issue.  This method is being told what extra register it should use when it needs one.  The caller has decided what the `initReg `should be and this method also writes back if it modifies the  initReg using `*pInitRegZeroed `

This method should instead assert that the `initReg` is a scratch register and the selection logic should avoid picking a callee save register.  

Another option is to remove the two parameters `regNumber initReg, bool* pInitRegZeroed,` from this method as it is not adding any real benefit here.

Reply:
It's ok to pick a callee-saved register as initReg for all other uses of initReg, which are all _after_ callee-saved regs have been saved. This function is also used by arm32, which uses initReg.

It's true that in my changed code, if we use REG_SCRATCH on arm64, we will still set `*pInitRegZeroed = false`, indicating that the passed-in initReg is trashed, which isn't strictly true, but isn't harmful. It turns out that arm64 _never_ sets `*pInitRegZeroed = true`, since if it needs a zero reg it uses `REG_ZR`.